### ISSUE TITLE
false alarm on `control_statement`?

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#5737]
### New Issue Checklist

- [x ] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

A clear and concise description of what the bug is. Ideally, provide a small (but compilable) example code snippet that
can be used to reproduce the issue.

```swift
            if foo == nil, (bar || foo) {
                bar = foo
            }
```

This was from periphery - the original code was

```
           if optBindingCondition.initializer == nil,
               let pattern = optBindingCondition.pattern.as(IdentifierPatternSyntax.self),
               let parentStmt = optBindingCondition.parent?.parent?.parent,
               (parentStmt.is(IfExprSyntax.self) || parentStmt.is(GuardStmtSyntax.self)) {
                // Handle `let x {}` syntax.
                parsed = parse(identifier: pattern.identifier)
            } else {
                parsed = parse(childrenFrom: node, collector)
            }
```

I don't think this should be triggering

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

I added this as an example to `triggeringViolations` and it passed

### Environment

* SwiftLint version (run `swiftlint version` to be sure)

0.56.1

* Xcode version (run `xcodebuild -version` to be sure)

Xcode 15.2

* Installation method used (Homebrew, CocoaPods, building from source, etc)
* Configuration file:

Running from source

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.

No

**Comment by SimplyDanny** [comment]
Why should it not trigger? Both

```swift
if foo == nil, (bar || foo) {
  bar = foo
}
```

and 

```swift
if foo == nil, bar || foo {
  bar = foo
}
```

are equivalent.

**Comment by mildm8nnered** [comment]
🤦
